ouroboros-consensus/docs/FastLaneSketch.md

```diff
@@ -0,0 +1,152 @@
+* the client https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-network/src/Ouroboros/Network/BlockFetch/Client.hs
+* the server https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/BlockFetch/Server.hs
+* the decision logic https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-network/src/Ouroboros/Network/BlockFetch.hs
+    * the node's sole logic thread is forked here at startup https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs#L158-L159
+    * the decision logic is ultimately implemented across multiple modules, many of which contain elaborate comments; these are the main ones
+        * https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs
+        * https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-network/src/Ouroboros/Network/BlockFetch/ClientState.hs
+        * https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-network/src/Ouroboros/Network/BlockFetch/State.hs
+    * I've mostly just left out the "fetch client registry", I think. I suggest understanding that part later.
+
+### ChainDB pointers
+
+* API https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/API.hs
+* the blocks-to-add queue https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Types.hs#L244
+* how BlockFetch adds blocks to the queue https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs#L333-L336
+* relevant background thread https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Background.hs#L523-L536
+* chain selection logic https://github.com/input-output-hk/ouroboros-network/blob/b0d95f33314e0562b6eea55455f1847efb85f8a4/ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs
+
+# Initial implementation plan
+
+I think I see a lightweight means of extending the above system to "rush"
+through the latest block and to do so only when it would matter.
+
+* Extend the ChainDB API with a part that notifies subscribed threads whenever
+  a new most-desirable-block-ever-seen is added.
```

Comment:
What is a new most-desirable-block-ever-seen? 

- How do we determine the desirability between two blocks? By their block numbers? 
- Most desirable among the blocks in the volatile DB?